### PR TITLE
Add --bg

### PR DESCRIPTION
**Pull request by kpcyrd**
This causes cjdroute client to exit, even though it was set to don't-background in the config.

This is --nobg's counterpart.

### CODE REVIEW

**Comment by lgierth** [comment]
:-1: 

I think instead of adding more complication around forking-to-background, we should just remove the noBackground config option.

**Comment by Kubuxu** [comment]
Background about this situation. As for now, systemd service file will assume that the service is up and running the moment it started (this is due to Type=simple). This breaks if some other service wants to bind IPs created by cjdns.

My change was https://github.com/hyperboria/cjdns/pull/117 but something in this style is much cleaner.

Removing noBackground from the config and allowing it to be only cmdline option would be probably the best solution.

@kpcyrd could you add systemd (and init) changes that would make sense with this option (so --bg and Type=forking).

**Comment by kpcyrd** [comment]
This patch removes noBackground now.

This is a breaking change if you don't use `--nobg` yet.

File: client/cjdroute2.c [hunk 1]
```diff
@@ -432,7 +423,7 @@ static int usage(struct Allocator* alloc, char* appName)
            "    cjdroute --bench               Run some cryptography performance benchmarks.\n"
            "    cjdroute --version             Print the protocol version which this node speaks.\n"
            "    cjdroute --cleanconf < conf    Print a clean (valid json) version of the config.\n"
-           "    cjdroute --nobg                Never fork to the background no matter the config.\n"
+           "    cjdroute --nobg                Don't fork to the background.\n"
```
Comment (Kubuxu) [inline client/cjdroute2.c]: Is there a space too much or is it GH trolling me.

File: client/cjdroute2.c [hunk 2]
```diff
@@ -730,13 +721,10 @@ int main(int argc, char** argv)
                         logger,
                         allocator);
 
-    // --------------------- noBackground ------------------------ //
-
-    int64_t* noBackground = Dict_getInt(&config, String_CONST("noBackground"));
-    if (forceNoBackground || (noBackground && *noBackground)) {
```
Comment (cjdelisle) [inline client/cjdroute2.c]: If I'm reading this right, upgrade is going to cause unintended effects for people who set nobg in the config file. +1 for removing it from `--genconf` though.

Comment (kpcyrd) [inline client/cjdroute2.c]: The initial change in 4e0d8ad15a280a564b818c57b94277b5af8d797b was quite complex, but was ensuring backward compatibility. If you'd prefer to merge that instead, I'd clean up this branch to `--nobg`, `--bg` and removing it from `--genconf`.

File: client/cjdroute2.c (hunk 1 above)
Comment (kpcyrd) [inline client/cjdroute2.c]: For me it's aligned on github and in my editor :)

**Comment by cjdelisle** [comment]
Merged to crashey, thanks

**Pull request closed without merging**